strings/decimal.c

```diff
@@ -493,7 +561,7 @@ static void digits_bounds(decimal_t *from, int *start_result, int *end_result)
     stop= (int) ((buf_end - from->buf + 1) * DIG_PER_DEC1);
     i= 1;
   }
-  for (; *buf_end % powers10[i++] == 0; stop--) ;
+  stop-= count_trailing_zeroes(i, *buf_end);
```

Comment:
This call seems to be suspicious - `*buf_end` can potentially be `0` here.
if the previous `while`
```cpp
while (buf_end > buf_beg  && *buf_end == 0)
     buf_end--;
```
terminated with `buf_end == buf_beg`

All the other calls to `count_trailing_zeroes` / `count_leading_zeroes` seems to guarantee that the second arg is non-zero.